kratos/solving_strategies/schemes/residual_based_bossak_displacement_scheme.hpp

```diff
@@ -899,199 +401,161 @@ class ResidualBasedBossakDisplacementScheme: public Scheme<TSparseSpace,TDenseSp
-
-            noalias(RHS_Contribution)  -= prod(M, mVector.a[thread]);
-            //KRATOS_WATCH( prod(M, mVector.a[thread] ) )
-
+        if (M.size1() != 0) {
+            Vector a, ap;
+            pElement->GetSecondDerivativesVector(a, 0);
+            a *= (1.00 - mAlpha.m);
+
+            pElement->GetSecondDerivativesVector(ap, 1);
+            noalias(a) += mAlpha.m * ap;
+            
+            noalias(RHS_Contribution) -= prod(M, a);
         }
 
         // Adding damping contribution
-        if (D.size1() != 0)
-        {
-            rCurrentElement->GetFirstDerivativesVector(mVector.v[thread], 0);
+        if (D.size1() != 0) {
+            Vector v;
+            pElement->GetFirstDerivativesVector(v, 0);
 
-            noalias(RHS_Contribution) -= prod(D, mVector.v[thread]);
+            noalias(RHS_Contribution) -= prod(D, v);
         }
     }
 
     /**
-     * It adds the dynamic RHS contribution of the condition: b - M*a - D*v
-     * @param rCurrentCondition: The condition to compute
-     * @param RHS_Contribution: The dynamic contribution for the RHS
-     * @param D: The damping matrix
-     * @param M: The mass matrix
-     * @param CurrentProcessInfo: The current process info instance
+     * @brief It adds the dynamic RHS contribution of the condition b - M*a - D*v
+     * @param pCondition The condition to compute
+     * @param RHS_Contribution The dynamic contribution for the RHS
+     * @param D The damping matrix
+     * @param M The mass matrix
+     * @param CurrentProcessInfo The current process info instance
      */
 
     void AddDynamicsToRHS(
-        Condition::Pointer rCurrentCondition,
+        Condition::Pointer pCondition,
         LocalSystemVectorType& RHS_Contribution,
         LocalSystemMatrixType& D,
         LocalSystemMatrixType& M,
-        ProcessInfo& CurrentProcessInfo)
+        ProcessInfo& CurrentProcessInfo
+        ) override
     {
-        int thread = OpenMPUtils::ThisThread();
-
         // Adding inertia contribution
-        if (M.size1() != 0)
-        {
-            rCurrentCondition->GetSecondDerivativesVector(mVector.a[thread], 0);
-
-            (mVector.a[thread]) *= (1.00 - mAlpha.m);
+        if (M.size1() != 0) {
+            Vector a, ap;
+            pCondition->GetSecondDerivativesVector(a, 0);
+            a *= (1.00 - mAlpha.m);
 
-            rCurrentCondition->GetSecondDerivativesVector(mVector.ap[thread], 1);
+            pCondition->GetSecondDerivativesVector(ap, 1);
+            noalias(a) += mAlpha.m * ap;
 
-            noalias(mVector.a[thread]) += mAlpha.m * mVector.ap[thread];
-
-            noalias(RHS_Contribution)  -= prod(M, mVector.a[thread]);
+            noalias(RHS_Contribution) -= prod(M, a);
         }
 
         // Adding damping contribution
         // Damping contribution
-        if (D.size1() != 0)
-        {
-            rCurrentCondition->GetFirstDerivativesVector(mVector.v[thread], 0);
```

Comment:
this was done this way to avoid reallocating. Why did u decide to change this???

Reply:
I will restore it, I didn't know it was because of that